docs: refactor readme + guidelines (#58)
arduino/arduino-app-cli: docs/contributor-guide/development.md

<!-- Source: https://github.com/arduino/tooling-project-assets/blob/main/documentation-templates/contributor-guide/other/development.md -->

# Development Guide

## Prerequisites

The following development tools must be available in your local environment:

- [Go](https://go.dev/dl/)
- [Taskfile](https://taskfile.dev/)

> - [ddb client]//https://developer.android.com/tools/adb) [optionally]

Reviewer: ```suggestion
- [ddb client](https://developer.android.com/tools/adb) [optionally]
```

Fix markup.

## Building the Project

- `task init`

## Uploading the arduino-app-cli into the board

> Connect an [Arduino UNO Q](https://www.arduino.cc/product-uno-q) board to the PC via USB.

Reviewer: ```suggestion
Connect an [Arduino UNO Q](https://docs.arduino.cc/hardware/uno-q/) board to the PC via USB.
```

From experience with the way the Arduino website is maintained, the previous URL is likely to stop working at some point in the future. The documentation pages are the most stable link targets (they typically only break when the URL changes after the product is retired).

>  - `task board:install` it installs the current `arduino-app-cli` inside the board (`adb` is needed). The password of the `ardiuno` username of the board is requested.

Reviewer: ```suggestion
 - `task board:install` installs the current version of Arduino App CLI on the board (`adb` is needed). The password of the `arduino` username of the board is requested.
```

- Grammar fixes
- Use official product name
- Correct username

<!-- Source: https://github.com/arduino/tooling-project-assets/blob/main/documentation-templates/contributor-guide/other/development.md -->

# Development Guide

> [!NOTE]
> The `arduino-app-cli` is designed to run on the Board and access peripherals that are not available on a development PC (e.g., the microcontroller).
>
> For easier testing, using an **Arduino UNO Q** is recommended, as local testing is limited to functionalities that do not require board-specific features.

## Prerequisites

The following development tools must be available in your local environment:

- [Go](https://go.dev/dl/)

> - [Task](https://taskfile.dev/)

Reviewer 2: You also need Docker to run most of the tests

<!-- Source: https://github.com/arduino/tooling-project-assets/blob/main/documentation-templates/contributor-guide/other/development.md -->

# Development Guide

> [!NOTE]
> The `arduino-app-cli` is designed to run on the Board and access peripherals that are not available on a development PC.
>
> For easier testing, using an **Arduino UNO Q** is recommended, as local testing is limited to functionalities that do not require board-specific features.

## Prerequisites

The following development tools must be available in your local environment:

- [Go](https://go.dev/dl/)
- [Task](https://taskfile.dev/)
- [Docker](https://docs.docker.com/engine/install/)
- [adb client](https://developer.android.com/tools/adb) [optionally]

## Building the Project

Build the project (run once):
- `task init`
- `task build`
- `task generate:assets` to download locally the assets of the [Arduino Bricks](`https://github.com/arduino/app-bricks-py`)

Start the arduino-app-cli in daemon mode:
- `ARDUINO_APP_CLI__DATA_DIR=debian/arduino-app-cli/home/arduino/.local/share/arduino-app-cli task start`

> NOTE: only a subset of HTTP APIs are working by running the daemon mode into a development PC. To run the `arduino-app-cli` into the board see `Running `arduino-app-cli` into the board` section.

Reviewer: ```suggestion
NOTE: only a subset of HTTP APIs are working by running the daemon mode on a development PC. To run Arduino App CLI on the board see the **Running Arduino App CLI on the board** section below.
```

- Correct grammar
- Use official product name
- Fix markup
